include/mbedtls/mbedtls_config.h

```diff
@@ -3680,8 +3680,56 @@
 
 /* Platform options */
 //#define MBEDTLS_PLATFORM_STD_MEM_HDR   <stdlib.h> /**< Header to include if MBEDTLS_PLATFORM_NO_STD_FUNCTIONS is defined. Don't define if no header is needed. */
-//#define MBEDTLS_PLATFORM_STD_CALLOC        calloc /**< Default allocator to use, can be undefined */
-//#define MBEDTLS_PLATFORM_STD_FREE            free /**< Default free to use, can be undefined */
+
+/* An overview of how the value of mbedtls_calloc is determined:
+ *
+ * if !MBEDTLS_PLATFORM_MEMORY
+ *     mbedtls_calloc = calloc
+ * if MBEDTLS_PLATFORM_MEMORY
+ *     if (MBEDTLS_PLATFORM_CALLOC_MACRO && MBEDTLS_PLATFORM_FREE_MACRO):
+ *         mbedtls_calloc = MBEDTLS_PLATFORM_CALLOC_MACRO
+ *     if !(MBEDTLS_PLATFORM_CALLOC_MACRO && MBEDTLS_PLATFORM_FREE_MACRO):
+ *         Dynamic setup via mbedtls_platform_set_calloc_free is now possible with a default value MBEDTLS_PLATFORM_STD_CALLOC.
+ *         How is MBEDTLS_PLATFORM_STD_CALLOC handled?
+ *         if MBEDTLS_PLATFORM_NO_STD_FUNCTIONS:
+ *             MBEDTLS_PLATFORM_STD_CALLOC is not set to anything;
+ *             MBEDTLS_PLATFORM_STD_MEM_HDR can be included if present;
+ *         if !MBEDTLS_PLATFORM_NO_STD_FUNCTIONS:
+ *             if MBEDTLS_PLATFORM_STD_CALLOC is present:
+ *                 User-defined MBEDTLS_PLATFORM_STD_CALLOC is respected;
+ *             if !MBEDTLS_PLATFORM_STD_CALLOC:
+ *                 MBEDTLS_PLATFORM_STD_CALLOC = calloc
+ *
+ *         At this point the presence of MBEDTLS_PLATFORM_STD_CALLOC is checked.
+ *         if !MBEDTLS_PLATFORM_STD_CALLOC
+ *             MBEDTLS_PLATFORM_STD_CALLOC = uninitialized_calloc
+ *
+ *         mbedtls_calloc = MBEDTLS_PLATFORM_STD_CALLOC.
+ *
+ * Defining MBEDTLS_PLATFORM_CALLOC_MACRO and MBEDTLS_PLATFORM_STD_CALLOC at the same time is not possible.
+ * MBEDTLS_PLATFORM_CALLOC_MACRO and MBEDTLS_PLATFORM_FREE_MACRO must both be defined or undefined at the same time.
+ * MBEDTLS_PLATFORM_STD_CALLOC and MBEDTLS_PLATFORM_STD_FREE do not have to be defined at the same time, as, if they are used,
+ * dynamic setup of these functions is possible. See the tree above to see how are they handled in all cases.
+ */
+
+/** \def MBEDTLS_PLATFORM_STD_CALLOC
+ *
+ * Default allocator to use, can be undefined. See the description above for details.
```

Comment:
“Above” is only meaningful when reading the source code, not when reading the result of the rendering by Doxygen.